Pptest

rourke750 opened the pull request:
Working branch of pp updates

rourke750 committed:
Initial changes to remove super method

rourke750 committed:
Fix minor conflict

rourke750 committed:
Moved summon and respawn methods to own class

Bunch of other minor fixes.

rourke750 committed:
Multiple fixes

ProgrammerDan commented on src/vg/civcraft/mc/prisonpearl/database/mysql/SummonMysqlStorage.java:
```diff
@@ -124,50 +124,7 @@ public void removeSummon(UUID uuid) {
-			}
-		}
-		return summon;
+		return summons.get(uuid);
```
I'm assuming then that we pre-stuff all the summons on load?

rourke750 replied:
yes.

ProgrammerDan commented on src/vg/civcraft/mc/prisonpearl/listener/PlayerListener.java:
```diff
@@ -429,7 +449,7 @@ else if(event.getAction() == InventoryAction.PLACE_ALL
 				InventoryHolder holder = !clickedTop ? event.getView().getTopInventory().getHolder() : event.getView().getBottomInventory().getHolder();
 				if (holder==null){
-					Player player=pearl.getHolderPlayer();
+					pearl.getHolderPlayer();
```
uh... isn't this supposed to be assigned to a thing? like, holder = pearl.getHolderPlayer() since it falls through to an additional check below? same QQ for line 423 above.

rourke750 replied:
ERm looks like legacy code.  It didnt do anything before as well.

ProgrammerDan replied:
Super weird :D

ProgrammerDan commented on src/vg/civcraft/mc/prisonpearl/managers/SummonManager.java:
```diff
@@ -90,18 +144,91 @@ public boolean returnPlayer(PrisonPearl pearl) {
+			// Let's check if the player is online.
+			if (pearled == null) {
+				// They are not so lets just remove the summon.
+				storage.removeSummon(s);
```
Just to check, this will result in them respawning back in the end when they next login?

rourke750 replied:
Yes. https://github.com/Civcraft/PrisonPearl/blob/d4ee27e96d821d879eb3ac8d902863cddf622907/src/vg/civcraft/mc/prisonpearl/listener/PlayerListener.java#L131
and then
https://github.com/Civcraft/PrisonPearl/blob/pptest/src/vg/civcraft/mc/prisonpearl/PrisonPearlUtilShards.java#L55

ProgrammerDan commented:
I think i see the framing here, good changes. I'll be around to help test.

rourke750 committed:
More fixes

Added more code in to utils classes.  Added methods for updating
summoned players accross shards.

rourke750 committed:
Update to for toggle commands

Toggle commands now persist for summoned players cross shards

rourke750 committed:
Fixed minor issue with ppl

rourke750 committed:
Minor fixes for returning and summoning

rourke750 commented:
@Maxopoly @ProgrammerDan Please review.  Testing has been done and waiting on merge.

Maxopoly commented on plugin.yml:
```diff
@@ -69,12 +69,7 @@ commands:
-      description: Silence reception of pplocate commands from player
-      usage: /ppsilence player
-      permission: prisonpearl.normal.ppsilence
+      permission: prisonpearl.normal.ppconfirm=
```
typo in the perm name

rourke750 replied:
Good catch.

rourke750 commented:
Any issues with me fixing that one problem, staging and merging?

Maxopoly commented:
Go for it

ProgrammerDan commented:
Uh, okay to delay an extra hour for me to review? <3

rourke750 commented:
@ProgrammerDan Still looking?

ProgrammerDan commented:
if you're confident via testing just go for it. I'd have to fine-tooth to really make sense of your changes, there are a lot. Ganbate, YOLO rourke-san.

rourke750 committed:
fix minor typo
